presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -339,7 +362,7 @@ private Expression unwrapCast(ComparisonExpression expression)
                         return new ComparisonExpression(LESS_THAN, cast.getExpression(), literalEncoder.toExpression(literalInSourceType, sourceType));
                     case GREATER_THAN:
                     case GREATER_THAN_OR_EQUAL:
-                        if (sourceRange.isPresent() && compare(sourceType, sourceRange.get().getMax(), literalInSourceType) == 0) {
+                        if (sourceRange.isPresent() && compare(sourceType, sourceRange.get().getMax(), literalInSourceType) == 0 && implicitInjectiveCoercionAbility == FULL) {
```

Comment:
The additional condition is redundant. For PARTIAL, the allowed range for literal here is well within source type range. Instead, please add a comment.